test/scope_token_to_list_of_private_public_repos_test.go

```diff
@@ -0,0 +1,160 @@
+
+	repositoryForPrivateRepo := &pacv1alpha1.Repository{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: targetNS,
+		},
+		Spec: pacv1alpha1.RepositorySpec{
+			URL: "https://github.com/" + splittedValue[3] + "/" + splittedValue[4],
+		},
+	}
+
+	err = trepo.CreateRepo(ctx, targetNS, runcnx, repositoryForPrivateRepo)
+	assert.NilError(t, err)
+
+	targetRefName := fmt.Sprintf("refs/heads/%s",
+		names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pac-e2e-test"))
+
+	sha, err := tgithub.PushFilesToRef(ctx, ghcnx.Client, "TestPullRequestRemoteAnnotations - "+targetRefName, repoinfo.GetDefaultBranch(), targetRefName, opts.Organization, opts.Repo, entries)
+	assert.NilError(t, err)
+	runcnx.Clients.Log.Infof("Commit %s has been created and pushed to %s", sha, targetRefName)
+
+	title := "TestPullRequestRemoteAnnotations - " + targetRefName
+	number, err := tgithub.PRCreate(ctx, runcnx, ghcnx, opts.Organization, opts.Repo, targetRefName, repoinfo.GetDefaultBranch(), title)
+	assert.NilError(t, err)
+
+	defer tgithub.TearDown(ctx, t, runcnx, ghcnx, number, targetRefName, targetNS, opts)
+
+	runcnx.Clients.Log.Infof("Waiting for Repository to be updated")
+	waitOpts := twait.Opts{
+		RepoName:        targetNS,
+		Namespace:       targetNS,
+		MinNumberStatus: 0,
+		PollTimeout:     twait.DefaultTimeout,
+		TargetSHA:       sha,
+	}
+	err = twait.UntilRepositoryUpdated(ctx, runcnx.Clients, waitOpts)
+	assert.NilError(t, err)
+
+	runcnx.Clients.Log.Infof("Check if we have the repository set as succeeded")
+	repo, err := runcnx.Clients.PipelineAsCode.PipelinesascodeV1alpha1().Repositories(targetNS).Get(ctx, targetNS, metav1.GetOptions{})
+	assert.NilError(t, err)
+	laststatus := repo.Status[len(repo.Status)-1]
+	assert.Equal(t, corev1.ConditionTrue, laststatus.Conditions[0].Status)
+	assert.Equal(t, sha, *laststatus.SHA)
+	assert.Equal(t, sha, filepath.Base(*laststatus.SHAURL))
+	assert.Equal(t, title, *laststatus.Title)
+	assert.Assert(t, *laststatus.LogURL != "")
+
+	pr, err := runcnx.Clients.Tekton.TektonV1().PipelineRuns(targetNS).Get(ctx, laststatus.PipelineRunName, metav1.GetOptions{})
+	assert.NilError(t, err)
+
+	assert.Equal(t, options.PullRequestEvent, pr.Labels["pipelinesascode.tekton.dev/event-type"])
+	assert.Equal(t, repo.GetName(), pr.Labels["pipelinesascode.tekton.dev/repository"])
+	assert.Equal(t, sha, pr.Labels["pipelinesascode.tekton.dev/sha"])
+	assert.Equal(t, opts.Organization, pr.Labels["pipelinesascode.tekton.dev/url-org"])
+	assert.Equal(t, opts.Repo, pr.Labels["pipelinesascode.tekton.dev/url-repository"])
+
+	assert.Equal(t, sha, filepath.Base(pr.Annotations["pipelinesascode.tekton.dev/sha-url"]))
+	assert.Equal(t, title, pr.Annotations["pipelinesascode.tekton.dev/sha-title"])
+	defer updateAndRevertCM(ctx, t, runcnx, "true")
+}
+
+func updateAndRevertCM(ctx context.Context, t *testing.T, runcnx *params.Run, value string) {
+	cmList, err := runcnx.Clients.Kube.CoreV1().ConfigMaps("").List(ctx, metav1.ListOptions{})
```

Comment:
why a List over a loop and not just a Get?

Reply:
Did list because i was not knowing that we are setting `SYSTEM_NAMESPAE` env in Setup function here https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/pkg/github/setup.go#L70

thats why did list in order to get ns

But now changed the code to do Get as we get ns info where pipelines-as-code cm is installed